src/pages/ExamsPage/hooks.js

```diff
@@ -172,6 +172,23 @@ export const useCreateAllowance = () => {
   };
 };
 
+export const useEditAllowance = () => {
+  const makeNetworkRequest = reduxHooks.useMakeNetworkRequest();
+  const courseId = useSelector(selectors.courseId);
+  const fetchAllowances = module.useFetchAllowances();
+  return (formData, closeModal) => {
+    const allowanceData = [formData];
+    makeNetworkRequest({
+      requestKey: RequestKeys.createAllowance,
+      promise: api.createAllowance(courseId, allowanceData),
+      onSuccess: () => {
+        closeModal();
+        fetchAllowances(courseId);
```

Comment:
I don't think we need a new backend request here (correct me if I'm wrong). We should just call a reducer function to edit state. As a example see how we modify an exam attempt rather than re-fetch the whole list.

Reply:
Cool, just updated!